src/dune_rules/context.ml

```diff
@@ -743,9 +743,10 @@ let instantiate_context env (workspace : Workspace.t)
       ; fdo_target_exe
       ; dynamically_linked_foreign_archives
       ; instrument_with
+      ; merlin
       } ->
     let merlin =
-      workspace.merlin_context = Some (Workspace.Context.name context)
+      workspace.merlin_context = Some (Workspace.Context.name context) || merlin
```

Comment:
Do we still need this? IIRC, the code that parses the workspace file in workspace.ml ensures that there is exactly one context that has `merlin` set to true.

Reply:
Yes it is still needed because the merlin context, when chosen by default, is not reflected in the context `merlin` field, only in `workspace.merlin_context`.

However the `|| merlin` part of the test is superfluous as if the user chose a context to enable merlin this should already be checked by the `workspace.merlin_context` test.